unit-tests/Test/Booster/Pattern/ApplyEquations.hs

```diff
@@ -36,29 +36,37 @@ test_evaluateFunction =
         "Evaluating functions using rules without side conditions"
         [ -- f1(a) => a
           testCase "Simple function evaluation" $ do
-            eval TopDown [trm| f1{}(A:SomeSort{}) |] @?= Right [trm| A:SomeSort{} |]
-            eval BottomUp [trm| f1{}(A:SomeSort{}) |] @?= Right [trm| A:SomeSort{} |]
-        , -- f2(f1(f1(a))) => f2(a). f2 is marked as partial, so not evaluating
+            eval TopDown [trm| f1{}(con2{}(A:SomeSort{})) |] @?= Right [trm| con2{}(A:SomeSort{}) |]
+            eval BottomUp [trm| f1{}(con2{}(A:SomeSort{})) |] @?= Right [trm| con2{}(A:SomeSort{}) |]
+        , -- f2(f1(f1(con2(a)))) => f2(con2(a)). f2 is marked as partial, so not evaluating
```

Comment:
i would have thought these should be indeterminate...
shouldn't any term with a variable return an indeterminate match??

Reply:
e.g. given

```
rule f(foo(bar())) => baz()
rule f(foo(A)) => A
```

where `foo`,`bar` and `baz` are constructors, we don't know what `f(foo(X))` should rewrite to

Reply:
That's kind-of what we had here, but it is not _any_ variable occurrence, just the variable being at the top matching the entire argument should be indeterminate.
The test fixture has rules 
```
f ( con1 ( X ) )                  => con2 ( f1 ( X ) )
f ( inj{..} ( con4 ( X , _Y ) ) ) => X
f ( X )                           => X
``` 
Previously  the test was using `f1 ( A )` (should be indeterminate indeed), now we use `f1 (con2 ( A ))`